### DISCUSSION TITLE
Update 22.11 -> 24.11; uptime is changed?

### DISCUSSION THREAD

**Discussion by luddite516** [NetApp/harvest#3402] [Q&A]
I am in the process of installing the latest harvest, and will be switching from ZAPI to REST-based data collection.

It appears that the metric 'node_uptime' which existed in 22.11 is not collected by REST.  Instead, I see 'uptime' as a label for the 'node_labels' object, and that should work for what we are using it for.

The problem I have is that the 'uptime' label does not appear in the node_labels objects which harvest REST has collected thus far.  The templates look good; I have not modified them.  But I am not understanding why all the node labels are being collected fine except for uptime?

Example of one of our node_labels instances:

node_labels{bmc_firmware_version="13.10P1", cluster="xxxxxxx", cpu_firmware_release="16.10", datacenter="xxxxxxx", ha_partner="xxxxxxx", healthy="true", instance="localhost:12007", job="xxxxxxx", location="xxxxxxx", max_aggr_size="879609302220800", max_vol_num="2500", max_vol_size="329853488332800", model="FAS8700", node="xxxxxxx", partner_healthy="true", serial="952347003575", site="test2", vendor="NetApp", version="NetApp Release 9.14.1P7: Thu Aug 01 03:07:44 UTC 2024", warnings="There are no failed fans.; There are no failed power supplies.; false"}

**Comment by rahulguptajss** [comment]
@luddite516 I see that the node_uptime metric is available in version 24.11 for both REST and ZAPI. `uptime` will not be part of `node_labels` as it is a metric in template, including it in labels could lead to cardinality issues in Prometheus. I have noticed that `uptime` is included in the `instance_labels` within the node template. Since `uptime` is a metric and not a label, its inclusion is redundant. We will remove it to avoid any confusion through the following pull request: https://github.com/NetApp/harvest/pull/3404.

Could you please email us the output of the API call `api/cluster/nodes?fields=*` for this cluster and Rest node template at [ng-harvest-files@netapp.com](mailto:ng-harvest-files@netapp.com)?

<img width="1913" alt="image" src="https://github.com/user-attachments/assets/08529bbe-e3d1-4f23-85a7-41a96793926c" />
